src/utils/makeReactNativeConfig.js

```diff
@@ -197,6 +213,10 @@ const getDefaultConfig = (
     mainFields: ['react-native', 'browser', 'main'],
     extensions: [`.${platform}.js`, '.native.js', '.js'],
   },
+  /**
+   * Set target to webworker as it's closer to RN environment than `web`.
+   */
+  target: 'webworker',
```

Comment:
`web` assumes you have the DOM, `webworker` doesn't have the DOM, so it uses different mechanism for downloading files - `importScripts` whereas `web` uses `script` tags.

Downloading hot update mechanism for `web`: https://github.com/webpack/webpack/blob/master/lib/JsonpMainTemplate.runtime.js#L12-L18

And the same for `webworker`: https://github.com/webpack/webpack/blob/master/lib/webworker/WebWorkerMainTemplate.runtime.js#L12-L14

React Native is closer to `webworker` in terms of environment than `web`.